File: StoreServer/src/main/java/com/example/demo/entities/LineaOrdine.java
package com.example.demo.entities;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Getter
@Setter
@ToString
@EqualsAndHashCode
@Entity
@Table(name = "lineaordine")
public class LineaOrdine {
    @Id
    @Column(name = "id", nullable = false)
    private long id;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "ordine")
    @ToString.Exclude
    private Ordine ordine;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "prodotto")
    private Prodotto prodotto;

    @Basic
    @Column(name = "qta")
    private int qta;

    @Basic
    @Column(name = "prezzo")
    private double prezzo;
}

File: StoreClient/src/app/services/lineaOrdine/lineaOrdine.ts
import { prodotto } from './../prodotto/prodotto';
import { ordine } from './../ordine/ordine';
export class lineaOrdine{
    id: number = 0; 
    ordine: ordine | null = null;
    prodotto: prodotto;
    qta: number;
    prezzo: number;

    constructor(p: prodotto, qta: number, prezzo: number){
        this.prodotto = p;
        this.qta = qta;
        this.prezzo = prezzo;
    }
}